homeassistant/components/ezviz/config_flow.py

```diff
@@ -303,11 +310,76 @@ async def async_step_confirm(self, user_input=None):
             data_schema=discovered_camera_schema,
             errors=errors,
             description_placeholders={
-                "serial": self.unique_id,
+                ATTR_SERIAL: self.unique_id,
                 CONF_IP_ADDRESS: self.context["data"][CONF_IP_ADDRESS],
             },
         )
 
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Handle a flow for reauthentication with password."""
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a Confirm flow for reauthentication with password."""
+        auth_data = {}
+        errors = {}
+        entry = None
+
+        for item in self._async_current_entries():
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                self.context["title_placeholders"] = {ATTR_SERIAL: item.title}
+                entry = await self.async_set_unique_id(item.title)
+
+        if not entry:
+            return self.async_abort(reason="ezviz_cloud_account_missing")
+
+        if user_input is not None:
+            user_input[CONF_URL] = entry.data[CONF_URL]
+
+            try:
+                auth_data = await self.hass.async_add_executor_job(
+                    _validate_and_create_auth, user_input
+                )
+
+            except (InvalidHost, InvalidURL):
+                errors["base"] = "invalid_host"
+
+            except EzvizAuthVerificationCode:
+                errors["base"] = "mfa_required"
+
+            except (PyEzvizError, AuthTestResultFailed):
+                errors["base"] = "invalid_auth"
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+            else:
+                self.hass.config_entries.async_update_entry(
+                    entry,
+                    data=auth_data,
+                )
+
+                await self.hass.config_entries.async_reload(entry.entry_id)
+
+                return self.async_abort(reason="reauth_successful")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=entry.title): vol.In([entry.title]),
```

Comment:
Same here.

Reply:
Addressed in https://github.com/home-assistant/core/pull/94384